django-backend/fecfiler/transactions/views.py

```diff
@@ -509,6 +672,59 @@ def remove_transaction_from_report(self, request):
         transaction.reports.remove(report)
         return Response("Transaction removed from report")
 
+    def process_general_election_year_aggregates(self, transaction_instance):
+        # Get the transaction out of the queryset in order to populate annotated fields
+        transaction = self.get_queryset().filter(id=transaction_instance.id).first()
+        if transaction is None:
+            return
+
+        queryset = self.get_queryset()
+
+        shared_entity_transactions = queryset.filter(
+            date__year=transaction.date.year,
+            contact_2=transaction.contact_2,
+            aggregation_group=transaction.aggregation_group,
+            schedule_f__isnull=False,
+            schedule_f__general_election_year=transaction.schedule_f.general_election_year,  # noqa: E501
+        )
+
+        previous_transactions = filter_for_previous_transactions(
+            shared_entity_transactions,
+            transaction.id,
+            transaction.date,
+            transaction.aggregation_group,
+            None,
+            transaction.contact_2.id,
+            None,
+            transaction.schedule_f.general_election_year
+        )
+
+        to_update = shared_entity_transactions.filter(
+            Q(id=transaction.id)
+            | Q(date__gt=transaction.date)
+            | Q(Q(date=transaction.date) & Q(created__gt=transaction.created))
+        ).order_by("date", "created")
+
+        previous_transaction = previous_transactions.first()
+        updated_schedule_fs = []
+        for trans in to_update:
+            previous_aggregate = 0
+            if previous_transaction:
+                previous_aggregate = (
+                    previous_transaction.schedule_f.aggregate_general_elec_expended
+                )
+
+            trans.schedule_f.aggregate_general_elec_expended = (
+                trans.schedule_f.expenditure_amount + previous_aggregate
+            )
+            updated_schedule_fs.append(trans.schedule_f)
+            previous_transaction = trans
+
+        ScheduleF.objects.bulk_update(
```

Comment:
I'm wondering if we might want to set the batch_size parameter in case we end up with an excessive number of previous transactions.  Thoughts?
https://docs.djangoproject.com/en/5.2/ref/models/querysets/#bulk-update

Reply:
It might be unnecessary because we're just altering one column, but I'm open to the idea.  Any thoughts on what a good size for the batch size would be?
![image](https://github.com/user-attachments/assets/8ea0635c-20bc-41ce-9fc0-68cf86accbed)